### PR TITLE
Implemented packet PACKET_ZC_GUILD_POSITION

### PR DESCRIPTION
**Pull request by Toshiro90**
* **Addressed Issue(s)**: -

* **Server Mode**: Both

* **Description of Pull Request**: 

Implemented packet `PACKET_ZC_GUILD_POSITION` (based on HerculesWS/Hercules@759065f).

### CODE REVIEW

**Review by aleos89 — APPROVED** [review]

File: src/map/clif.cpp [hunk 1]
```diff
@@ -9347,6 +9347,30 @@ void clif_guild_broken( map_session_data& sd, int flag ){
...
+
+	p->packetType = HEADER_ZC_GUILD_POSITION;
+	p->packetLength = sizeof( PACKET_ZC_GUILD_POSITION );
+	p->AID = sd.bl.id;
+
+	if( sd.guild != nullptr ){
+		const auto& g = sd.guild->guild;
+
+		if( int ps = guild_getposition( sd ); ps != -1 ){
+			safestrncpy( p->position, g.position[ps].name, 24 );
```
Comment (AoShinRO) [inline src/map/clif.cpp:9363]:
```suggestion
			safestrncpy( p->position, g.position[ps].name, NAME_LENGTH );
```

File: src/map/clif.cpp [hunk 2]
```diff
@@ -9347,6 +9347,30 @@ void clif_guild_broken( map_session_data& sd, int flag ){
...
+			p->packetLength += static_cast<decltype(p->packetLength)>( 24 );
```
Comment (AoShinRO) [inline src/map/clif.cpp:9364]:
```suggestion
			p->packetLength += static_cast<decltype(p->packetLength)>( NAME_LENGTH );
```

File: src/map/guild.cpp [hunk 3]
```diff
@@ -1528,7 +1528,7 @@ int guild_position_changed(int guild_id,int idx,struct guild_position *p) {
 	// Update char name in client [LuzZza]
```
Comment (AoShinRO) [inline src/map/guild.cpp:1528]:
```suggestion
	// Update char name in client
```

File: src/map/guild.cpp [hunk 4]
```diff
@@ -1497,7 +1497,7 @@ int guild_memberposition_changed(struct mmo_guild &g,int idx,int pos) {
 
 	// Update char position in client [LuzZza]
```
Comment (AoShinRO) [inline src/map/guild.cpp:1498]:
```suggestion
	// Update char position in client
```

File: src/map/clif.cpp (hunk 2 above)
Comment (AoShinRO) [reply src/map/clif.cpp:9364]: funny thing, all others position info packets have position name var as char name[NAME_LENGTH]

File: src/map/clif.cpp [hunk 5]
```diff
@@ -9347,6 +9347,30 @@ void clif_guild_broken( map_session_data& sd, int flag ){
...
+	p->packetType = HEADER_ZC_GUILD_POSITION;
+	p->packetLength = sizeof( PACKET_ZC_GUILD_POSITION );
+	p->AID = sd.bl.id;
+
+	if( sd.guild != nullptr ){
+		const auto& g = sd.guild->guild;
+
+		if( int ps = guild_getposition( sd ); ps != -1 ){
+			safestrncpy( p->position, g.position[ps].name, sizeof(g.position[ps].name) );
+			p->packetLength += static_cast<decltype(p->packetLength)>( sizeof(g.position[ps].name) );
```
Comment (Lemongrass3110) [inline src/map/clif.cpp:9364]:
```suggestion
			safestrncpy( p->position, g.position[ps].name, NAME_LENGTH );
			p->packetLength += static_cast<decltype(p->packetLength)>( NAME_LENGTH );
```

**Review by Lemongrass3110 — APPROVED** [review]

**Pull request merged**